common/src/proof.rs

```diff
@@ -113,34 +117,77 @@ impl StarkProof {
 // ================================================================================================
 
 impl Queries {
-    /// Returns a set of queries constructed from a batch Merkle proof and corresponding values.
-    pub fn new<E: FieldElement>(merkle_proof: BatchMerkleProof, values: Vec<Vec<E>>) -> Self {
+    /// Returns a set of queries constructed from a batch Merkle proof and corresponding elements.
+    pub fn new<E: FieldElement>(merkle_proof: BatchMerkleProof, query_values: Vec<Vec<E>>) -> Self {
+        assert!(!query_values.is_empty(), "query values cannot be empty");
+        let elements_per_query = query_values[0].len();
+        assert!(
+            elements_per_query != 0,
```

Comment:
prefer `is_empty`

Reply:
I usually do too - but in this case we need to cache the `elements_per_query` value - so, I think it's OK to have it like this.